cli/tests/include.bats

```diff
@@ -0,0 +1,172 @@
+  pushd "$PROJECT_DIR" >/dev/null || return
+  export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.json"
+}
+
+project_teardown() {
+  popd >/dev/null || return
+  rm -rf "${PROJECT_DIR?}"
+  rm -f "${TMP_MANIFEST_PATH?}"
+  unset PROJECT_DIR
+  unset MANIFEST_PATH
+  unset TMP_MANIFEST_PATH
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  setup_isolated_flox
+  project_setup
+  export _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/empty.json"
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+
+# ---------------------------------------------------------------------------- #
+#
+# NOTE: The following functionality is tested elsewhere:
+#
+#   - logs: providers/services.rs
+#   - status: providers/services.rs
+#
+# ---------------------------------------------------------------------------- #
+
+setup_composer_and_two_includes() {
+  # Setup included1 environment
+  "$FLOX_BIN" init -d included1
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [vars]
+    included1 = "v1"
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f - -d included1
+
+  # Setup included2 environment
+  "$FLOX_BIN" init -d included2
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [vars]
+    included2 = "v1"
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f - -d included2
+
+  # Setup composer
+  "$FLOX_BIN" init -d composer
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [include]
+    environments = [
+      { dir = "../included1" },
+      { dir = "../included2" },
+    ]
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f - -d composer
+}
+
+# Modify vars.included1 in environment included1
+edit_included1() {
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [vars]
+    included1 = "v2"
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f - -d included1
+}
+
+edit_both_included_environments() {
+  edit_included1
+
+  # Edit included2
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [vars]
+    included2 = "v2"
+EOF
+  )"
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f - -d included2
+
+}
+
+@test "include upgrade reports no changes" {
+  setup_composer_and_two_includes
+  run "$FLOX_BIN" include upgrade -d composer
+  assert_success
+  assert_output "ℹ️ No included environments have changes."
+}
+
+@test "include upgrade reports no changes when non-upgraded environment changes" {
+  setup_composer_and_two_includes
+  edit_included1
+  run "$FLOX_BIN" include upgrade -d composer included2
+  assert_success
+  assert_output "ℹ️ Included environment 'included2' has no changes."
+}
+
+@test "include upgrade defaults to upgrading all" {
+  setup_composer_and_two_includes
+  edit_both_included_environments
+
+  run "$FLOX_BIN" include upgrade -d composer
+  assert_success
+  assert_output - <<EOF
+✅ Upgraded included environment 'included1'
+✅ Upgraded included environment 'included2'
```

Comment:
followup: I think saying `Upgraded included environment` makes it sound like we're upgrading the included environment in the `flox upgrade` sense. I think this is another reason we need a better term than `included environment`. @jennymahmoudi thoughts?

Reply:
Not jenny but thoughts nonetheless:
I think syntactically this literally says "upgraded ... environment", while it should say "upgraded include [of environment ...]" (the use of "include" as a noun may be debatable).

Reply:
nb: Also we have been moving to printing a single message with a list of subjects for install and upgrade, i thjink we should do the same here

Reply:
Starting a thread in Slack